### ISSUE TITLE
Replace remaining uses of underscore.js with lodash

### ISSUE THREAD

**Issue by camjackson** [rabblerouser/core#25]
Lodash is _almost_ a superset of underscore.js. So we should be able to just replace all references to underscore with equivalent uses of lodash. But whoever picks this up should double check that nothing breaks as we swap one out for the other.
- [x] Replace frontend uses of underscore.js with equivalent lodash
- [x] Remove underscore.js dependency from `frontend/package.json`
- [x] Remove underscore.js dependency from frontend shrinkwrap
- [x] Replace backend uses of underscore.js with equivalent lodash
- [x] Remove underscore.js dependency from `backend/package.json`
- [x] Remove underscore.js dependency from backend shrinkwrap

Reference for previous discussion: https://stackoverflow.com/questions/13789618/differences-between-lodash-and-underscore

**Comment by akjones** [comment]
+1 lodash

**Comment by rdoh** [comment]
+2

**Comment by pameck** [comment]
+1000

**Comment by pameck** [comment]
So, Lodash? Do we close it? Or leave it open until we remove underscore from wherever it is?

**Comment by camjackson** [comment]
Let's repurpose this issue - it's a good beginner one.

**Comment by camjackson** [comment]
@phazel If you want a beginner issue, this might be a good way to get your hands on some code? I'd be happy to help you get started.

**Comment by intersim** [comment]
@camjackson I can help on this if you still need it!

**Comment by camjackson** [comment]
Sure thing @intersim! Let us know if you have any questions or need help getting started!

**Comment by intersim** [comment]
@camjackson I'm having some issues getting rabblerouser-core set up on my machine... just running `npm install` hangs when it gets to installing Jade, but then when I run each of the install scripts separately and then run the build, it says it can't find webpack, even though it's in the frontend node modules...? anything I'm missing?

```
> rabblerouser-core-frontend@0.1.0 build /vagrant/frontend
> webpack --progress --colors 

sh: 1: webpack: not found

npm ERR! Linux 3.13.0-91-generic
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "--prefix" "frontend" "run" "build"
npm ERR! node v4.4.7
npm ERR! npm  v2.15.8
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn
npm ERR! rabblerouser-core-frontend@0.1.0 build: `webpack --progress --colors `
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the rabblerouser-core-frontend@0.1.0 build script 'webpack --progress --colors'.
```

**Comment by pameck** [comment]
Hi @intersim !
I just re-install everything from scratch to try to reproduce the error, but it didn't happen to me, I install everything with a single `npm install`, it took me 25mins, but it installed all the packages.

When I run: `npm --prefix frontend run build` from /vagrant it builds the app.

If you list the node_modules folder in frontend, what do you get? This is what I get:

``` bash
$ ls /vagrant/frontend/node_modules/ | grep webpack
extract-text-webpack-plugin
karma-webpack
webpack
webpack-dev-middleware
webpack-dev-server
```

**Comment by camjackson** [comment]
Hey @intersim! Firstly, thanks for getting involved with the project! Hopefully we can fix your error so you can focus on the codes :)

This error: `sh: 1: webpack: not found` looks like a shell error (rather than a node/npm one), telling you that the webpack executable is not on your PATH. Which means that the command `webpack --progress --colors` will fail.

The location of that executable should be `rabblerouser-core/frontend/node_modules/.bin/webpack`. That `.bin` directory is a special location for node module executables, and npm adds it to your path automatically whenever you do `npm run some-script`. However, this feature has not always been there in npm, so I wonder if you're running an old version of node/npm? Are you using the vagrant dev setup that we provide? Could you give us the output of these commands: `node -v` and `npm -v`

In the case that you do have that executable in the location mentioned above, you could try changing the build task in the `frontend/package.json` to include the full relative path:

```
...
  "build": "./node_modules/.bin/webpack --progress --colors",
...
```

**Comment by intersim** [comment]
@pameck @camjackson thanks for the responses, I'll give it another shot!

**Comment by camjackson** [comment]
Did you have any luck @intersim? Is there anything we can do to help?

**Comment by intersim** [comment]
@camjackson hey all! Sorry, I ended up getting a new job and forgot all about this... I'll give it another shot over the weekend and report back.

**Comment by camjackson** [comment]
No problem at all! Let us know how you go. Just a quick note that @yearofthedan is currently tweaking some of our dev setup stuff. If you already set up the vagrant box previously, you may need to blow it away with `vagrant destroy` and start again.

**Comment by intersim** [comment]
@camjackson ah ok, good to know!

**Comment by intersim** [comment]
Still working on getting my dev environment working... destroyed the old box, now `vagrant up` is resulting in

`VBoxManage: error: The configured driver wasn't found. Either the necessary driver modules wasn't loaded, the name was misspelled, or it was a misconfiguration. (VERR_PDM_DRIVER_NOT_FOUND)
VBoxManage: error: Details: code NS_ERROR_FAILURE (0x80004005), component ConsoleWrap, interface IConsole`

@camjackson @yearofthedan

**Comment by camjackson** [comment]
Hmm, do you have VirtualBox installed?

Also, what's your host OS? Windows/OSX/Linux/etc?

**Comment by intersim** [comment]
@camjackson yup, I have VirtualBox, and I'm on OSX. Also, I went in and disabled audio for this box on VirtualBox (to try to address the earlier error), and now I'm getting this out when I run `vagrant up`:

```
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Checking if box 'bento/ubuntu-16.04' is up to date...
==> default: Running provisioner: ansible...
    default: Running ansible-playbook...
[DEPRECATION WARNING]: Instead of sudo/sudo_user, use become/become_user and 
make sure become_method is 'sudo' (default).
This feature will be removed in a 
future release. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.

PLAY [all] *********************************************************************

TASK [setup] *******************************************************************
fatal: [default]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh.", "unreachable": true}
    to retry, use: --limit @provisioning/dev.retry

PLAY RECAP *********************************************************************
default                    : ok=0    changed=0    unreachable=1    failed=0   

Ansible failed to complete successfully. Any error output should be
visible above. Please fix these errors and try again.
```

**Comment by intersim** [comment]
I'm going to keep trying to get my dev environment working correctly, but if anyone also wants to tackle this ticket, please feel free!

**Comment by camjackson** [comment]
Not exactly sure what's causing that error message, but it looks like you can't SSH into the VM. It's not a great solution, but we've seen that be caused by firewalls in the past. Maybe try disabling yours?

**Comment by yearofthedan** [comment]
Fixed in #122
